artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/etc/artemis-utility.profile

```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+ARTEMIS_HOME='${artemis.home}'
+ARTEMIS_INSTANCE='@artemis.instance@'
+ARTEMIS_DATA_DIR='${artemis.instance.data}'
+ARTEMIS_ETC_DIR='${artemis.instance.etc}'
+ARTEMIS_OOME_DUMP='${artemis.instance.oome.dump}'
+
+# The logging config will need an URI
+# this will be encoded in case you use spaces or special characters
+# on your directory structure
+ARTEMIS_INSTANCE_URI='${artemis.instance.uri}'
+ARTEMIS_INSTANCE_ETC_URI='${artemis.instance.etc.uri}'
+
+# Java Opts
+if [ -z "$JAVA_ARGS" ]; then
+    JAVA_ARGS="-Dlog4j2.configurationFile=${ARTEMIS_INSTANCE_ETC_URI}log4j2-utility.properties ${java-opts}"
+fi
```

Comment:
The trouble with using JAVA_ARGS for this directly is, it then wont have any effect for anyone that is already setting JAVA_ARGS now, unless they notice this was added and adjust their JAVA_ARGS to include it. It also means anyone setting JAVA_ARGS later has to set this too, when they probably dont want to influence the logging.

Between that, and people perhaps thinking they need to replace JAVA_ARGS to change this (if they dont know later definitions take priority and e.g use JAVA_ARGS_APPEND to set it 'again'), I think it would be worth considering having a specific variable for the logging config file selection to use (if set; the main profile wouldnt set it, this profile would) such that people could use it to override/set what the profile chooses to pass on to the main script. Much like the new profile picker variable. People could then still use JAVA_ARGS or JAVA_ARGS_APPEND to override it too if they want.

(As example, part of the docs you updated in this PR, suggested people to set JAVA_ARGS for a command, but doesnt include setting this, so it wouldnt use this new logging config in that case)

Reply:
+1

Reply:
It sounds good, I'll add the variable `LOGGING_ARGS` before `JAVA_ARGS`